tck/tck-runtime/src/main/java/servlet/tck/spec/httpservletresponse/HttpTestServlet.java

```diff
@@ -42,22 +45,44 @@ public void intHeaderTest(HttpServletRequest request,
     response.addIntHeader("header2", 56789);
   }
 
-  public void flushBufferTest(HttpServletRequest request,
+  public void flushBufferOnContentLengthTest(HttpServletRequest request,
       HttpServletResponse response) throws ServletException, IOException {
     int size = 40;
     response.setContentLength(size);
 
-    PrintWriter pw = response.getWriter();
-    pw.println("Test PASSED");
-    StringBuffer tmp = new StringBuffer(2 * size);
-    int i = 0;
-
-    while (i < 8) {
-      tmp = tmp.append("111111111x");
-      i = i + 1;
-    }
-    pw.println(tmp);
+    OutputStream out = response.getOutputStream();
+    byte[] passed = "Test PASSED\n".getBytes(StandardCharsets.ISO_8859_1);
+    out.write(passed);
+
+    byte[] fill = new byte[size - passed.length];
+    Arrays.fill(fill, (byte) 'x');
+    out.write(fill);
     response.addIntHeader("header1", 12345);
+
+    try {
+      out.write(fill);
+      throw new IllegalStateException("write did not fail");
```

Comment:
Is this sufficient to trigger a test failure?

Reply:
It should be as the requirement is that the proceeding write closes the output stream. Wiring to a closed output stream should be an exception regardless of buffering.

Reply:
We could add an out.flush if you are really concerned, but I don't see it as necessary

Reply:
Sorry, should have been clearer. I'm not doubting the exception will be thrown. My question is whether the client will see any effects of the exception which will in turn triggered a test failure. Or to put it another way, will the client see the completed response and think all is well regardless of whether the exception is thrown on the server.

Reply:
In this test, the write a line 59 will complete the response and close the stream.  The client will see that response as a 200.     The subsequent write will throw an exception.... ah I see the problem, the client will not see the exception either way.  Let me ponder a better approach....  hard in this test environment as everything is sent via the response.

Reply:
@olamy Can you test the latest version please

Reply:
@gregw all good with jetty 12.1.x

Reply:
@markt-asf How's this version?  It does every test request twice with any bad behaviour after response completion of the first request stored in the session to be seen by the second request.

Reply:
Neat solution. LGTM.